### ISSUE TITLE
Python 2.7 compatibility issues with natsort 7.0+

### ISSUE THREAD

**Issue by lowell80** [taichino/croniter#149]
Just as a heads up, I ran into a Python 2.7 compatibility issue with the `natsort` library.  Currently croniter doesn't specify a specific version requirements, just its presence.  This has lead to some runtime import errors, specifically:

```ImportError: No module named compat.locale```

I've worked around this in my own project by adding the following to my `requirements.txt` file:
```
natsort>=6.0,<7.0
```

I see that natsorted notes this deprecation here:  https://pypi.org/project/natsort/#deprecation-schedule

It seems like this is only an issue if you actually build a package (which contains an exported library) using Python 3 rather than Python 2.   Maybe that's an unusual use case?

**Comment by lowell80** [comment]
Upon additional inspection of #127, I'm wondering if a simpler solution to the Python 3 stricter data type comparison may be possible without an additional external library.  By using the `key` argument in `sorted()` this could be pretty easily workaround this issue by casting values to strings

```python
>>> res = [7,14,21,"L"]
>>> sorted(res, key=lambda i: "{:02}".format(i) if isinstance(i, int) else i)
[7, 14, 21, 'L']
```

Which produces the same results as:

```python
>>> from natsort import natsorted
>>> natsorted(res)
[7, 14, 21, 'L']
```

I've confirmed that this works for Python 2 and 3, but haven't attempted to run unittests against it.   Would you be willing to consider a PR that removes this external dependency using an approach like this?

**Comment by kiorky** [comment]
Yep, proceed with unittests.

Also test `10, 1, 1111, 111, la, l` -> `1,10,1111,111,l,la`

**Comment by lowell80** [comment]
@kiorky, just getting back around to this, thanks for your patience.  I'll have a PR shortly.  All existing unittests seems to be working as expected.

I didn't turn your comment into a unittests, but copied it as an example here.   I did add a new test for validating that sorting works as expected in `expanded`.

```python
>>> from natsort import natsorted
>>> res = [10, 1, '1111', '11', '111', "la","l"]
>>> sorted(res, key=lambda i: "{:02}".format(i) if isinstance(i, int) else i)
[1, 10, '11', '111', '1111', 'l', 'la']
>>> natsorted(res)
[1, 10, '11', '111', '1111', 'l', 'la']
```

The only place `natsorted` was being used was where integers cannot exceed double digits.  So I think the `{:02}` should be fine.

BTW, is `0 0 * * *,sat#3` a legitimate cron expression?  I copied it into the new unittest, because I identified it as an example of where a normal `sorted()` doesn't work, but I'm not sure why how it's any different from `0 0 * * *`  (it seems like `*` plus anything else could always just be optimized down to simply `*`).  Anyways, if that's a legitimate expression, that's great, otherwise it can be removed.
